components/login/provider/index.jsx

```diff
@@ -19,76 +19,139 @@
  * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
+/* eslint react/jsx-props-no-spreading: 0 */
+
 import React, { useState } from "react";
-import { Box, TextField } from "@material-ui/core";
+import T from "prop-types";
+import {
+  Box,
+  FormControl,
+  FormHelperText,
+  TextField,
+  useTheme,
+} from "@material-ui/core";
 import { Autocomplete } from "@material-ui/lab";
 
-import { createStyles, makeStyles } from "@material-ui/styles";
-import { useBem } from "@solid/lit-prism-patterns";
 import { LoginButton, useSession } from "@inrupt/solid-ui-react";
 
+import { Button } from "@inrupt/prism-react-components";
 import { generateRedirectUrl } from "../../../src/windowHelpers";
-
 import getIdentityProviders from "../../../constants/provider";
-import styles from "./styles";
 
 const providers = getIdentityProviders();
 const TESTCAFE_ID_LOGIN_TITLE = "login-title";
+export const TESTCAFE_ID_LOGIN_FIELD = "login-field";
 const TESTCAFE_ID_LOGIN_BUTTON = "login-button";
 
-const useStyles = makeStyles((theme) => createStyles(styles(theme)));
+export function setupOnProviderChange(setProviderIri) {
+  return (e, newValue) => {
+    setProviderIri(newValue);
+  };
+}
+
+export function setupLoginHandler(login) {
+  return async (event) => {
+    event.preventDefault();
+    login();
+  };
+}
 
-export default function Provider() {
-  const bem = useBem(useStyles());
+export function setupErrorHandler(setLoginError) {
+  return (error) => {
+    setLoginError(error);
+  };
+}
+
+export function getErrorMessage(error) {
+  const postFix = " Please fill out a valid Solid Identity Provider.";
+  if (
+    error.message.match(/fetch/g) || // Chrome, Edge, Firefox
```

Comment:
These regexes (and everything below) should be in constants. When you're checking multiple, use an array and do a `.find`:

`if(URL_NOT_IDP_REGEXES.find(r => error.message.match(r)) { /* stuff */ }`

Or even clean it up with a reusable function:

`if(hasError(URP_NOT_IDP_REGEXES, error)) { /* stuff */ }`

Reply:
I introduced `src/error` as a new module in the latest commit, and introduced your suggestions there.